telluric/util/local_tile_server.py

```diff
@@ -0,0 +1,86 @@
+import asyncio
+import folium
+import rasterio
+import tornado.web
+from tornado import gen
+from threading import Thread, Lock
+from tornado.ioloop import IOLoop
+from telluric.constants import WGS84_CRS
+
+
+class TileServerHandler(tornado.web.RequestHandler):
+    def initialize(self, rasters):
+        self.rasters = rasters
+
+    @tornado.web.asynchronous
+    @gen.coroutine
+    def get(self, raster_id, x, y, z):
+        with rasterio.Env():
+            tile = self.rasters[int(raster_id)].get_tile(int(x), int(y), int(z))
+        self.write(tile.to_png())
+        self.set_header("Content-type", "image/png")
+        self.finish()
+
+
+class OKHandler(tornado.web.RequestHandler):
+    def get(self):
+        self.write("i'm alive")
+
+
+def make_app(rasters):
+    uri = r'/(\d+)/(\d+)/(\d+)/(\d+)\.png'
+    return tornado.web.Application([
+        (uri, TileServerHandler, dict(rasters=rasters)),
+        (r'/ok', OKHandler),
+    ])
+
+
+def _run_app(rasters, port=4000):
+    asyncio.set_event_loop(asyncio.new_event_loop())
+    app = make_app(rasters)
+    app.listen(port, '0.0.0.0')
+    IOLoop.current().start()
+
+
+rasters_lock = Lock()
+
+
+class TileServer:
+    rasters = {}
+    running_app = None
+
+    @classmethod
+    def folium_client(cls, raster, base_map="Stamen Terrain", port=4000, zoom_start=13):
```

Comment:
zoom_start should be calculated, it could be that for some raster is too low

Reply:
resolved in new code